Sync/Synchronizers/ClientsSynchronizer.swift

```diff
@@ -417,11 +418,18 @@ open class ClientsSynchronizer: TimestampedSingleCollectionSynchronizer, Synchro
         // the fact that `fxaDeviceId` may not have been populated if the list of clients
         // hadn't changed since before the update to v8.0. To force a re-sync, we get all
         // clients since the beginning of time instead of looking at `self.lastFetched`.
+        let isFirstSync = self.lastFetched == 0
```

Comment:
I think this logic should probably live near where `maybeUploadOurClientRecord` is called. That's the point we decide whether to upload our record, and we know why — someone sent us a tab, it's more than a week since we uploaded, etc. etc.